src/main/java/cn/edu/tsinghua/tsfile/timeseries/write/io/TsFileIOWriter.java

```diff
@@ -0,0 +1,239 @@
+package cn.edu.tsinghua.tsfile.timeseries.write.io;
+
+import cn.edu.tsinghua.tsfile.common.conf.TSFileDescriptor;
+import cn.edu.tsinghua.tsfile.common.utils.BytesUtils;
+import cn.edu.tsinghua.tsfile.common.utils.ListByteArrayOutputStream;
+import cn.edu.tsinghua.tsfile.common.utils.TsRandomAccessFileWriter;
+import cn.edu.tsinghua.tsfile.common.utils.ITsRandomAccessFileWriter;
+import cn.edu.tsinghua.tsfile.file.metadata.*;
+import cn.edu.tsinghua.tsfile.file.metadata.converter.TSFileMetaDataConverter;
+import cn.edu.tsinghua.tsfile.file.metadata.enums.CompressionTypeName;
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSChunkType;
+import cn.edu.tsinghua.tsfile.file.metadata.enums.TSDataType;
+import cn.edu.tsinghua.tsfile.file.metadata.statistics.Statistics;
+import cn.edu.tsinghua.tsfile.file.utils.ReadWriteThriftFormatUtils;
+import cn.edu.tsinghua.tsfile.timeseries.write.desc.MeasurementDescriptor;
+import cn.edu.tsinghua.tsfile.timeseries.write.schema.FileSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * TSFileIOWriter is used to construct metadata and write data stored in memory to output stream.
+ *
+ * @author kangrong
+ */
+public class TsFileIOWriter {
+  public static final String MAGIC_STRING = "TsFilev0.0.1";
```

Comment:
The magic_string are not consistent with the actual version.

Reply:
I do not think this string needs to keep up with our software version. 
Actually,  it only needs to be changed when our serialization method is changed.